lib/internal/buffer_new.js

```diff
@@ -103,7 +103,10 @@ function fromObject(obj) {
     return b;
   }
 
-  // TODO(trevnorris): This will fail for an actual ArrayBuffer.
+  if (obj instanceof ArrayBuffer) {
```

Comment:
Nit: in theory this can be fooled (causing crashes) with things like `{ __proto__: ArrayBuffer }` which is not really an ArrayBuffer. This kind of thing is prevalent throughout io.js of course, but I know you were looking at related stuff.